source/constructing-lattices.md

```diff
@@ -0,0 +1,306 @@
+be defined "in place". Example:
+```{code} yaml
+BeamLine:
+  line:
+    - item: inj_line          # Refer by name to a previously defined BeamLine
+    - item:
+        BeamLine:             # Define a subline in place
+          line:
+           ...
+    ...
+```
+
+Restriction: Infinite recursion of sublines is not allowed. 
+For example, if BeamLine `B` is a subline of `A`, then BeamLine `A` may not be a subline of `B`.
+
+%---------------------------------------------------------------------------------------------------
+(s:repetition)=
+### Repetition
+
+For any line `item`, a `repetition` count component can be used to represent multiple copies
+of the item. Example:
+```{code} yaml
+BeamLine:
+  name: full_line
+  line:
+    - item:
+        name: short_line
+        repetition: 3
+```
+In this case, `short_line` is repeated three times when the BeamLine is expanded to form a lattice
+branch. For example, if `short_line` is defined by:
+```{code} yaml
+BeamLine:
+  name: short_line
+  line:
+    - item: A
+    - item: B
+    - item: C
+```
+then the expanded `full_line` will look like:
+```{code} yaml
+A, B, C, A, B, C, A, B, C
+```
+
+repetition counts can be negative. In this case, the elements are taken to occur in reverse order.
+Thus, in the above example, if the `repetition` count was `-3`, the expanded `full_line` will
+look like:
+```{code} yaml
+C, B, A, C, B, A, C, B, A
+```
+
+Notice that reverse order does not mean true [direction reversal](#s:ref.construct). 
+For elements that have longitudinal symmetry, this does not matter. 
+However, for example, for a `Bend` element, that is in a line with reversed order,
+the edge angle `e1` will still represent the edge of the upstream side and `e2` will represent the edge 
+at the downstream side.
+
+%---------------------------------------------------------------------------------------------------
+(s:direction)=
+### Direction reversal
+
+The optional `direction` component of `item` can be used for true [direction reversal](#s:ref.construct).
+Possible values are `+1` and `-1`. The Default is `+1` which represents an unreversed element
+or BeamLine. BeamLine reversal involves both reversed order of the line and direction reversal of
+the individual line items. Example:
+```{code} yaml
+BeamLine:
+  name: lineA
+  line:
+    - item: 
+        direction: -1
+        name: lineB
+
+BeamLine:
+  name: lineB
+  line:
+    - item: ele1
+    - item:
+        direction: -1
+        name: ele2
+```
+The expanded `lineA` has elements:
+```{code} yaml
+ele2, -ele1
+```
+where the negative sign here indicates that `ele1` is reversed in direction. 
+Notice that `ele2` is not reversed since the reversal of a reversed element results
+in an element that is unreversed.
+
+%---------------------------------------------------------------------------------------------------
+(s:placement)=
+### Line item placement
+
```

Comment:
I am confused by this section, even with the figure. To discuss at the PALS meeting.

Reply:
Discussion indicates this is resolved..